testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -111,7 +111,12 @@ impl Case for BlockchainTestCase {
             // Validate post state
             match &case.post_state {
                 Some(RootOrState::Root(root)) => {
-                    // TODO: We should really check the state root here...
+                    let last_block_number = last_block.unwrap();
+                    let header = provider.tx_ref().get::<tables::Headers>(last_block_number)?.ok_or_else(|| {
+                        Error::Assertion(format!("Expected header for block number ({last_block_number:?}) is missing from DB: {self:?}"))
+                    })?;
+                    let actual_root = header.state_root;
+                    assert_eq!(root, &actual_root);
                     println!("Post-state root: #{root:?}")
```

Comment:
Do we still need this print statement as soon as there is an equality assertion just above and therefore any abnormal behavior will necessarily be returned?

Reply:
Yes actually it could be deleted. I wait for some other feedback and if it's ok I can delete it then